strapi/strapi
# Fix deep populate issue

### What does it do?

Fixes deep populate issue with a wildcard notation

### Why is it needed?

It's a regression introduced by a previous PR

### How to test it?

- Create a dynamic zone with at least two components, add relational field in at least one of the components
- attempt to fetch with following query `http://localhost:1337/api/countries?populate[dynamiczone][populate]=*`
- relational fields should be populated

### Related issue(s)/PR(s)

fixes #17654
